elements/pfe-readtime/pfe-readtime.ts

```diff
@@ -0,0 +1,171 @@
+      return 228;
+    case 'zh': // 158 wpm
+      return 158;
+    case 'fr': // 195 wpm
+      return 195;
+    case 'ja': // 193 wpm
+      return 193;
+    case 'de':
+      return 179;
+    case 'it': // 188 wpm
+      return 188;
+    case 'pt-br': // 181 wpm
+      return 181;
+    case 'es':
+      return 218;
+    default:
+      return 228;
+  }
+}
+
+/**
+ *
+ * This component takes in the word count of a given section and does a calculation on that number to return an estimated read time based on language. The words-per-minute values were sourced from [this article](https://irisreading.com/average-reading-speed-in-various-languages) with the data originating from [this research](https://iovs.arvojournals.org/article.aspx?articleid=2166061). For more information, see the [Readtime calculation information](#readtime-calculation).
+ */
+@customElement('pfe-readtime') @pfelement()
+export class PfeReadtime extends LitElement {
+  static readonly styles = [style];
+
+  /**
+   * Allows you to manually set the number of words to use in the readtime calculations. Example: `word-count="2500"` will set the component to use 2500 words in it's calculations.
+   */
+  @property({ type: Number, reflect: true, attribute: 'word-count' }) wordCount = 0
+
+  /**
+   * Is the attribute used to store the average words per minute readtime for each supported country. For more information on these you can read [https://irisreading.com/average-reading-speed-in-various-languages](https://irisreading.com/average-reading-speed-in-various-languages) and [https://iovs.arvojournals.org/article.aspx?articleid=216606](https://iovs.arvojournals.org/article.aspx?articleid=216606).
+   */
+  @property({ type: Number, reflect: true }) wpm?: number = getEstimatedWPM(this.lang);
+
+  /**
+   * Rather than use the light DOM region to provide the string template, you can also pass in the value via this attribute. Note that %t will be replaced with the computed readtime.
+   * Template for printing the Readtime
+   * Translatable string for printing out the readtime in a readable format.
+   * Use %t as a stand-in for the calculated value.
+   */
+  @property({ attribute: 'template', reflect: true })
+  templateString = this.textContent?.trim() || '%t-minute readtime';
+
+  /**
+   * By default the component will look to the language specified on the html tag but it can also accept an override via this attribute on a component level.
+   */
+  @observed
+  @property({ reflect: true })
+  lang: string = document.documentElement.lang || 'en';
+
+  /**
+   * Specify the selector of the content you want to capture the word-count from.  Whatever you enter into this attribute will be found using `document.querySelector(<for attribute value>)`.
+   */
+  @observed
+  @property({ type: String, reflect: true }) for?: string
+
+  @property({ type: Number, reflect: true }) readtime = 0;
```

Comment:
`readonly` flag in CEM is still [just a proposal](https://github.com/webcomponents/custom-elements-manifest/issues/34) but I don't mind using it in a non-standard way.

If the property _is_ readonly, would it make more sense to write it as a getter, and set the attribute imperatively when the dependencies change, rather than decorating a class field with `@property`?

Reply:
Nice, yeah I agree.  I'll submit a PR for that. :)

Reply:
you can add [cem-plugin-readonly](https://github.com/bennypowers/cem-plugins/tree/main/plugins/cem-plugin-readonly) as a dependency to pfe-tools package (not repo root!) to get the CEM support